saltproc/_schema_default.py

```diff
@@ -0,0 +1,21 @@
+from jsonschema import Draft202012Validator, validators
+
+def extend_with_default(validator_class):
+    validate_properties = validator_class.VALIDATORS["properties"]
+
+    def set_defaults(validator, properties, instance, schema):
+        for property, subschema in properties.items():
+            if "default" in subschema:
+                instance.setdefault(property, subschema["default"])
+
+        for error in validate_properties(
+            validator, properties, instance, schema,
+        ):
+            yield error
+
+    return validators.extend(
+        validator_class, {"properties" : set_defaults},
+    )
```

Comment:
I didn't catch this before, but what is the purpose of having nested function definitions? Why not break this into two functions that can both be unit-tested?

Reply:
I'm not sure. I pulled this block of code from [here](https://python-jsonschema.readthedocs.io/en/stable/faq/#why-doesn-t-my-schema-s-default-property-set-the-default-on-my-instance)

Reply:
@samgdotson I actually created [an issue](https://github.com/arfc/saltproc/issues/184) to look into this.